### PR TITLE
Add loggercheck linter to verify that \*w logging methods have even number of args.

### PR DESCRIPTION
**Pull request by pavel-raykov**

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

File: .golangci.yml
Comment (jmank88) [inline .golangci.yml:1]: There is also code in this repo that uses the logger from chainlink-common, so we should copy that method set here too.

Comment (pavel-raykov) [reply .golangci.yml:1]: Thanks Jordan - this allowed us to caught a ~ dozen more issues in the code.

**Review by jmank88 — DISMISSED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -47,6 +48,19 @@ linters-settings:
           - (github.com/smartcontractkit/chainlink/v2/core/logger.Logger).Criticalf
           - (github.com/smartcontractkit/chainlink/v2/core/logger.Logger).Panicf
           - (github.com/smartcontractkit/chainlink/v2/core/logger.Logger).Fatalf
+          - (github.com/smartcontractkit/chainlink/v2/core/logger.SugaredLogger).AssumptionViolationf
```
Comment (jmank88) [inline .golangci.yml:51]: Dupe
```suggestion
```

Comment (jmank88) [reply .golangci.yml:51]: Er nvm

**Review by jmank88 — DISMISSED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by george-dorin — APPROVED** [review]

**Pull request merged**